packages/actions/src/helpers/database.ts

```diff
@@ -123,6 +124,26 @@ export const getDocumentById = async (
     return getDoc(docRef)
 }
 
+export const waitForUserDocumentToExist = (firestoreDatabase: Firestore, collection: string, documentId: string) => {
```

Comment:
while this would be a great solution, the code might hang forever if a user tried to auth and they did not pass the sybil checks. As discussed, a fetch and retry mechanism might work better for now until we find a more elegant solution